### PR TITLE
configure and document aggregation limits

### PR DESCRIPTION
**Pull request by PSeitz**
### Description

Add 2 options on the searcher config:
- aggregation_memory_limit 
- aggregation_bucket_limit

create and share AggregationLimits
fix yaml config test
simplify default deserialization
update documentation

### How was this PR tested?

Describe how you tested this PR.

```
➜  quickwit-indices curl -X POST -w '\nTotal: %{time_total}s\n' -H "Content-Type: application/json" -d @agg_request.json  http://127.0.0.1:7280/api/v1/air_mail/search
{"message":"Internal error: `(Internal error: `Aborting aggregation because memory limit was exceeded. Limit: 100.00 KB, Current: 100.35 KB`., split_id: 01GWKPM4XEQ46QF2R95Z05YA24), (Internal error: `Aborting aggregation beca
use memory limit was exceeded. Limit: 100.00 KB, Current: 100.35 KB`., split_id: 01GWKPNZH3SQCWZKVASF7H6DPY), (Internal error: `Aborting aggregation because memory limit was exceeded. Limit: 100.00 KB, Current: 100.35 KB`., s
plit_id: 01GWKPQT4YSZ7YGJM0QAF7GEF7), (Internal error: `Aborting aggregation because memory limit was exceeded. Limit: 100.00 KB, Current: 100.35 KB`., split_id: 01GWKPSMR86Z54N71CB5852DQT), (Internal error: `Aborting aggrega
tion because memory limit was exceeded. Limit: 100.00 KB, Current: 100.35 KB`., split_id: 01GWKPVFBJZQVAVNB02JJCME98), (Internal error: `Aborting aggregation because memory limit was exceeded. Limit: 100.00 KB, Current: 100.3
5 KB`., split_id: 01GWKPX9YTZZC25WF6S769TYDS), (Internal error: `Aborting aggregation because memory limit was exceeded. Limit: 100.00 KB, Current: 100.35 KB`., split_id: 01GWKPZ4JNH0375BZG44XSRXZE)`."}
Total: 0.043159s
```

### CODE REVIEW

File: docs/configuration/node-config.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ This section contains the configuration options for a Searcher.
 
 | Property | Description | Default value |
 | --- | --- | --- |
+| `aggregation_memory_limit` | Controls the maximum amount of memory that can be used for aggregations before aborting. This limit is shared among all queries on splits on a node for a single query. It is used to prevent excessive memory usage during the aggregation phase, which can lead to performance degradation or crashes. | `500M`|
+| `aggregation_bucket_limit` | Determines the maximum number of buckets returned to the client. | `65000` |
```
Comment (fmassot) [inline docs/configuration/node-config.md:69]: does it abort if the maximum number of buckets exceeds the limit?

**Review by fmassot — APPROVED** [review]
See my comment. It could be nice to add a test with a very low limit so that we are sure it will fail and check that.

File: docs/configuration/node-config.md (hunk 1 above)
Comment (PSeitz) [reply docs/configuration/node-config.md:69]: The check is done after the aggregations, before returning to the client

File: docs/configuration/node-config.md [hunk 2]
```diff
@@ -65,6 +65,8 @@ This section contains the configuration options for a Searcher.
 
 | Property | Description | Default value |
 | --- | --- | --- |
+| `aggregation_memory_limit` | Controls the maximum amount of memory that can be used for aggregations before aborting. This limit is shared among all queries on splits on a node for a single query. It is used to prevent excessive memory usage during the aggregation phase, which can lead to performance degradation or crashes. | `500M`|
```
[the remark this answers is not in the document]
Comment (PSeitz) [reply docs/configuration/node-config.md]: yes that's better

**Comment by PSeitz** [comment]
> It could be nice to add a test with a very low limit so that we are sure it will fail and check that.

The tests currently use single_node_search, I can add one after https://github.com/quickwit-oss/quickwit/issues/2765

**Comment by fmassot** [comment]
> > It could be nice to add a test with a very low limit so that we are sure it will fail and check that.
> 
> The tests currently use single_node_search, I can add one after #2765

yes, we can open an issue to not forget it

**Pull request merged**